lib/versioner/versioner.go

```diff
@@ -8,11 +8,13 @@
 // simple default versioning scheme.
 package versioner
 
+import "github.com/syncthing/syncthing/lib/fs"
+
 type Versioner interface {
 	Archive(filePath string) error
 }
 
-var Factories = map[string]func(folderID string, folderDir string, params map[string]string) Versioner{}
+var Factories = map[string]func(folderID string, filesystem fs.Filesystem, params map[string]string) Versioner{}
```

Comment:
Does this mean the versioning directory needs to live in the same filesystem as the folder, that is, underneath the folder root? I'm not necessarily opposed to that (it's even hardcoded into everything but the staggered thing) but it is a change